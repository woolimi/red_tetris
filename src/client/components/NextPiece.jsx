import React from "react";
import Cell from "./Cell";
import styled from "styled-components";
import { TETROMINOS } from "../gameHelper";

const NextPieceContainer = styled.div`
	box-sizing: border-box;
	margin-top: 3em;
	margin-bottom: 1em;
	padding: 20px;
	border: 3px solid #333;
	background: #000;
	font-family: Pixel, Arial, Helvetica, sans-serif;
	font-size: 1rem;
	align-items: center;
	width: 100%;
	border-radius: 20px;
`;

const StyledNextPiece = styled.div`
	display: grid;
	grid-template-columns: repeat(${(props) => props.columns}, 1fr);
	grid-gap: 1px;
	width: 5vw;
	height: 5vw;
	margin: auto;
`;

const NextPiece = ({ type }) => {
	return (
		<NextPieceContainer>
			<div>NEXT</div>
			<StyledNextPiece columns={type && TETROMINOS[type].matrix[0].length}>
				{type &&
					TETROMINOS[type].matrix.map((row) =>
						row.map((cell, x) => <Cell key={x} type={cell} />),
					)}
			</StyledNextPiece>
		</NextPieceContainer>
	);
};

export default React.memo(NextPiece);
