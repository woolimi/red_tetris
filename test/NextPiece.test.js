import React from "react";
import NextPiece from "../src/client/components/NextPiece";
import { render } from "@testing-library/react";

describe("<NextPiece />", () => {
	it("shows next piece I", () => {
		let type = "I";
		const { container } = render(<NextPiece type={type} />);
		expect(container).toMatchSnapshot();
	});
	it("shows next piece J", () => {
		let type = "J";
		const { container } = render(<NextPiece type={type} />);
		expect(container).toMatchSnapshot();
	});
	it("shows next piece L", () => {
		let type = "L";
		const { container } = render(<NextPiece type={type} />);
		expect(container).toMatchSnapshot();
	});
	it("shows next piece O", () => {
		let type = "O";
		const { container } = render(<NextPiece type={type} />);
		expect(container).toMatchSnapshot();
	});
	it("shows next piece S", () => {
		let type = "S";
		const { container } = render(<NextPiece type={type} />);
		expect(container).toMatchSnapshot();
	});
	it("shows next piece T", () => {
		let type = "T";
		const { container } = render(<NextPiece type={type} />);
		expect(container).toMatchSnapshot();
	});
	it("shows next piece Z", () => {
		let type = "Z";
		const { container } = render(<NextPiece type={type} />);
		expect(container).toMatchSnapshot();
	});
});
