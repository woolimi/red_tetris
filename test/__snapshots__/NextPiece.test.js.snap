// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NextPiece /> shows next piece I 1`] = `
<div>
  <div
    class="sc-gtssRu cwZlDc"
  >
    <div>
      NEXT
    </div>
    <div
      class="sc-dlnjPT kcElVS"
    >
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx fESRnj"
        color="80,227,230"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx fESRnj"
        color="80,227,230"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx fESRnj"
        color="80,227,230"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx fESRnj"
        color="80,227,230"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
    </div>
  </div>
</div>
`;

exports[`<NextPiece /> shows next piece J 1`] = `
<div>
  <div
    class="sc-gtssRu cwZlDc"
  >
    <div>
      NEXT
    </div>
    <div
      class="sc-dlnjPT LCkch"
    >
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx iORPbL"
        color="36,95,223"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx iORPbL"
        color="36,95,223"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx iORPbL"
        color="36,95,223"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx iORPbL"
        color="36,95,223"
        data-testid="cell"
        type="N"
      />
    </div>
  </div>
</div>
`;

exports[`<NextPiece /> shows next piece L 1`] = `
<div>
  <div
    class="sc-gtssRu cwZlDc"
  >
    <div>
      NEXT
    </div>
    <div
      class="sc-dlnjPT LCkch"
    >
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx fhXknm"
        color="223,173,36"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx fhXknm"
        color="223,173,36"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx fhXknm"
        color="223,173,36"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx fhXknm"
        color="223,173,36"
        data-testid="cell"
        type="N"
      />
    </div>
  </div>
</div>
`;

exports[`<NextPiece /> shows next piece O 1`] = `
<div>
  <div
    class="sc-gtssRu cwZlDc"
  >
    <div>
      NEXT
    </div>
    <div
      class="sc-dlnjPT kcElVS"
    >
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx kvGAfT"
        color="223,217,36"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx kvGAfT"
        color="223,217,36"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx kvGAfT"
        color="223,217,36"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx kvGAfT"
        color="223,217,36"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
    </div>
  </div>
</div>
`;

exports[`<NextPiece /> shows next piece S 1`] = `
<div>
  <div
    class="sc-gtssRu cwZlDc"
  >
    <div>
      NEXT
    </div>
    <div
      class="sc-dlnjPT LCkch"
    >
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx brHyQg"
        color="48,211,56"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx brHyQg"
        color="48,211,56"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx brHyQg"
        color="48,211,56"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx brHyQg"
        color="48,211,56"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
    </div>
  </div>
</div>
`;

exports[`<NextPiece /> shows next piece T 1`] = `
<div>
  <div
    class="sc-gtssRu cwZlDc"
  >
    <div>
      NEXT
    </div>
    <div
      class="sc-dlnjPT LCkch"
    >
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx fPvjwm"
        color="132,61,198"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx fPvjwm"
        color="132,61,198"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx fPvjwm"
        color="132,61,198"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx fPvjwm"
        color="132,61,198"
        data-testid="cell"
        type="N"
      />
    </div>
  </div>
</div>
`;

exports[`<NextPiece /> shows next piece Z 1`] = `
<div>
  <div
    class="sc-gtssRu cwZlDc"
  >
    <div>
      NEXT
    </div>
    <div
      class="sc-dlnjPT LCkch"
    >
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx hHoHXG"
        color="227,78,78"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx hHoHXG"
        color="227,78,78"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx hHoHXG"
        color="227,78,78"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx hHoHXG"
        color="227,78,78"
        data-testid="cell"
        type="N"
      />
    </div>
  </div>
</div>
`;

exports[`<NextPiece /> shows proper next piece depends on block type 1`] = `
<div>
  <div
    class="sc-gtssRu cwZlDc"
  >
    <div>
      NEXT
    </div>
    <div
      class="sc-dlnjPT LCkch"
    >
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx fPvjwm"
        color="132,61,198"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx dRVgAB"
        color="0,0,0"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx fPvjwm"
        color="132,61,198"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx fPvjwm"
        color="132,61,198"
        data-testid="cell"
        type="N"
      />
      <div
        class="sc-bdnylx fPvjwm"
        color="132,61,198"
        data-testid="cell"
        type="N"
      />
    </div>
  </div>
</div>
`;
